toolchain/lex/tokenized_buffer.cpp

```diff
@@ -894,6 +900,42 @@ class [[clang::internal_linkage]] TokenizedBuffer::Lexer {
          .string_id = buffer_.value_stores_->strings().Add(identifier_text)});
   }
 
+  auto LexKeywordOrIdentifierMaybeRaw(llvm::StringRef source_text,
+                                      ssize_t& position) -> LexResult {
+    CARBON_CHECK(source_text[position] == 'r');
+    // Raw identifiers must look like `r#<valid identifier>`, otherwise it's an
+    // identifier starting with the 'r'.
+    // TODO: Need to add support for Unicode lexing.
+    if (position + 2 >= static_cast<ssize_t>(source_text.size()) ||
```

Comment:
I wonder if it's worth putting an `LLVM_LIKELY` annotation on this `if` condition, or perhaps on the second part -- character 2 is unlikely to be `#`.

Reply:
Done.